Preparing Validation Operators and Actions for 0.8.0

eugmandel opened the pull request:
* Updated the signature of the run method of the operators.
* GE includes the following two operators: PerformActionListValidationOperator and RunWarningAndFailureExpectationSuitesValidationOperator. Only PerformActionListValidationOperator is included in the context's configuration by default.
* All actions are accepting a single "validation result suite"
* GE currently includes the actions: SlackNotificationAction, StoreAction and ExtractAndStoreEvaluationParamsAction
* This PR includes the PR with the updated signature of get_batch

abegong committed:
Deprecate register_validation_result and replace with DataContext.validate; Make minimal changes for dc.validate to function

abegong committed:
Modify profile_datasource to store profiling results; Drop a test with deprecated behavior

abegong committed:
Add data_context.add_validation_operator; Add BasicDataContextAwareValidationOperator; Enable dot-notation in action keys

abegong committed:
Debug and improve tests

roblim committed:
Fix index page cell width

eugmandel committed:
Replaced the usage of the 'name' attribute that was deprecated with 'class_name'; started printing expectation suites for each asset to make the method more informative

eugmandel committed:
WIP: Preparing the Validation Operators and Actions for the 0.8.0 release:
Modified the signature of the run method of the operators.
Updated the implementation of an operator that runs all validation results through a configured list of actions.
Updated the actions classes and added more actions.

roblim committed:
Add Slack action to init

roblim committed:
Fix typo

roblim committed:
Fix typo

roblim committed:
Add SlackNotifcationAction

roblim committed:
Add SlackNotificationAction test

roblim committed:
Add SlackRenderer

roblim committed:
Add SlackRenderer to init

roblim committed:
Resolve merge conflicts

jcampbell committed:
Update get_batch api

eugmandel committed:
WIP: updated the errors/warnings validation operator + some cleanup

roblim committed:
Add notifications section to template

roblim committed:
Nix slack notification callback logic from data context

roblim committed:
Nix slack functions in util in favor of action

roblim committed:
Add docstring and clean up SlackNotificationAction

roblim committed:
Merge branch 'eugene/val_ops_080' into feature/actions

roblim committed:
Add test for SlackRenderer

roblim committed:
Add test for SlackNotificationAction

roblim committed:
Change method name to “run”; fix typos

abegong reviewed (commented):
@eugmandel This direction makes sense to me. The gist of each one makes sense.

It would be really helpful to see ~3 example configs for each class. The main usability tradeoff is  flexibility vs conciseness in the configs. A few examples in tests would be really helpful for getting a sense for how this will feel to users.

eugmandel committed:
Added data_asset_identifiers key to return object of ErrorsVsWarningsValidationOperator

eugmandel committed:
multiple fixes an test updates

roblim committed:
resolve merge conflicts

roblim committed:
Bring send_slack_notification into util

roblim committed:
Implement slack notification in ErrorsVsWarningsValidationOperator

roblim committed:
Update test

roblim committed:
Add test for FailureVsWarningValidationOperator

eugmandel committed:
Merge pull request #729 from great-expectations/feature/actions

Feature/actions

jcampbell committed:
Update get_batch_api

eugmandel committed:
Merge branch 'eugene/val_ops_080' into feature/get_batch_api_update_alt

jcampbell committed:
Update get_batch_api tests

eugmandel committed:
Include only one validation operator in the context config template

jcampbell committed:
Merge branch 'feature/get_batch_api_update_alt' of github.com:great-expectations/great_expectations into feature/get_batch_api_update_alt

jcampbell committed:
Deprecate _get_or_convert_batch

jcampbell committed:
Add docstring

jcampbell committed:
Update create_expectations notebook

eugmandel committed:
tests

roblim committed:
Nix notification section

jcampbell committed:
Merge remote-tracking branch 'origin/eugene/val_ops_080' into feature/get_batch_api_update_alt

roblim committed:
Fix typo

roblim committed:
Update docstring

roblim committed:
Update action to not pull slack webhook from context config

roblim committed:
Add spaces

roblim committed:
Update test

roblim committed:
Update slack renderer with docs url

roblim committed:
Add notify_on functionality

roblim committed:
Update slack message with link to docs; update formatting

jcampbell committed:
Update validation operator usage

roblim committed:
Add test for slack query in test_errors_warnings_validation_operator_run_slack_query

roblim committed:
Comment out test_perform_action_list_validation_operator_run

jcampbell committed:
Update validation operator test to demonstrate get_batch param usage

roblim committed:
Fix merge conflicts

eugmandel committed:
Merge pull request #732 from great-expectations/feature/actions

Updated slack messages; slack tests

jcampbell committed:
Add profiling store to support data context-managed profiling

jcampbell committed:
Do not require data_asset_identifier property on batch

jcampbell committed:
Merge remote-tracking branch 'origin/eugene/val_ops_080' into feature/get_batch_api_update_alt

eugmandel committed:
Removed some mentions of the save shapshot feature

eugmandel committed:
Merge pull request #730 from great-expectations/feature/get_batch_api_update_alt

Feature/get batch api update alt

eugmandel committed:
Merge remote-tracking branch 'origin/eugene/val_ops_080' into eugene/val_ops_080

eugmandel committed:
test fix

eugmandel committed:
Renamed FailureVsWarningValidationOperator to RunWarningAndFailureExpectationSuitesValidationOperator

eugmandel committed:
Merge remote-tracking branch 'origin/v0.8.0_prep' into eugene/val_ops_080

eugmandel committed:
test fix

eugmandel committed:
test fix

eugmandel committed:
1. Fix: use the config with variables substituted when initializing validation operators
2. Fix: the method that substitues variables in config did not expect values to be lists and the recursion did not go in. Fixed.

jcampbell committed:
Add profiling_store_name to test_preserve_comments

jcampbell committed:
Begin refactoring documentation

jcampbell committed:
Rename-focused refactor

jcampbell committed:
Major documentation updates

eugmandel committed:
Merge pull request #733 from great-expectations/feature/docs_refactor

Feature/docs refactor

eugmandel committed:
Improved the docstrings of the operators, removed the tests that were commented out and replaced after the operator changes

roblim committed:
Fix bug that was breaking imports of great_expectations.jupyter_ux

- unused default function arguments of the form:

section_renderer=render.renderer.column_section_renderer.ExpectationSuiteColumnSectionRenderer,
view_renderer=render.view.view.DefaultJinjaSectionView,

were breaking imports

jcampbell committed:
Documentation cleanup and addition of several new stubs

jcampbell committed:
More significant docs refactors

eugmandel committed:
Merge pull request #735 from great-expectations/fix/jupyter-ux

Fix bug that was breaking imports of great_expectations.jupyter_ux

eugmandel committed:
Merge branch 'v0.8.0_prep' into eugene/val_ops_080

eugmandel committed:
Merge remote-tracking branch 'origin/feature/docs_refactor' into eugene/val_ops_080

eugmandel committed:
merge conflict resolution
